File: web/src/main/java/net/danil/web/model/ProblemLanguage.java
package net.danil.web.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ProblemLanguage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private Problem problem;

    @OneToMany(mappedBy = "problemLanguage", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Solution> solutions;

    @Enumerated(EnumType.STRING)
    private Language language;

    private String template;
    private String test;
}

File: web/src/main/java/net/danil/web/domain/ProblemLanguage.kt
package net.danil.web.domain

import jakarta.persistence.*

@Entity
@Table(name = "problem_language")
open class ProblemLanguage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    lateinit var problem: Problem

    @ManyToOne(fetch = FetchType.LAZY)
    lateinit var language: Language

    var initialSolution: String = ""

    var preloaded: String = ""

    var tests: String = ""
}
